cmd/nfd-topology-updater/main.go

```diff
@@ -86,6 +113,10 @@ func parseArgs(flags *flag.FlagSet, osArgs ...string) (*topology.Args, *resource
 		os.Exit(2)
 	}
 
+	if len(resourcemonitorArgs.KubeletConfigURL) == 0 {
+		resourcemonitorArgs.KubeletConfigURL = fmt.Sprintf("https://%s:%d/configz", nfdclient.NodeName(), kubeletSecurePort)
+	}
```

Comment:
Would `https://127.0.0.1:%d/configz` work and be safer bet?

Another suggestion, how about putting this as default value in `initFlags()` (`flagset.StringVar(...`) ?

Reply:
> Would https://127.0.0.1:%d/configz work and be safer bet?

I'm afraid `127.0.0.1` does not work unless `nfd-topology-updater` is host network, right? Kubelet is in host network namespace while `nfd-topology-updater` is in container network namespace. 

Meanwhile, you may be curious about why not using https. As you say, could we use ca from `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt` where we get the token by default? Actually, we cannot, let me explain.

Kubelet provides a TLS configz interface. Normally, we could retrieve the config from calling `proxy` subresource of `corev1.Node`. You could test it by using `curl`.

1. `nohup kubectl proxy &`
2. `curl http://127.0.0.1:8001/api/v1/nodes/{NODE_NAME}/proxy/configz`

So, if we curl configz interface from apiserver, actually the ca file is used for interact with apiserver. And apiserver use a different ca file which is normally identical with `/var/lib/kubelet/pki/kubelet.crt`, which is generated by kubeadm. In my minikube environment, they are totally different.

Our `nfd-topology-updater` does not have a connection between apiserver, it tries to connect `nfd-master` and send topology data, which updates NRT indirectly. I decide to use insecure interface for 2 reasons.

1. If use https, we should mount kubelet cert file in the container. However, this file is really confusing with apiserver ca file.
2. The connection between `nfd-topology-updater` and kubelet is localhost, all traffic will not run through internet, so it's ok to use insecure connection.